support/k8s/k8s-workload-registrar/controllers/pod_controller.go

```diff
@@ -0,0 +1,221 @@
+)
+
+type PodReconcilerMode int32
+
+const (
+	PodReconcilerModeServiceAccount PodReconcilerMode = iota
+	PodReconcilerModeLabel
+	PodReconcilerModeAnnotation
+)
+
+// PodReconcilerConfig holds the config passed in when creating the reconciler
+type PodReconcilerConfig struct {
+	Log                logrus.FieldLogger
+	Mgr                ctrl.Manager
+	TrustDomain        string
+	PodLabel           string
+	PodAnnotation      string
+	DisabledNamespaces []string
+}
+
+// PodReconciler holds the runtime configuration and state of this controller
+type PodReconciler struct {
+	client.Client
+	c              PodReconcilerConfig
+	Scheme         *runtime.Scheme
+	Mode           PodReconcilerMode
+	Value          string
+}
+
+// NewPodReconciler creates a new PodReconciler object
+func NewPodReconciler(config PodReconcilerConfig) (*PodReconciler, error) {
+	mode := PodReconcilerModeServiceAccount
+	value := ""
+	if len(config.PodLabel) > 0 {
+		mode = PodReconcilerModeLabel
+		value = config.PodLabel
+	}
+	if len(config.PodAnnotation) > 0 {
+		mode = PodReconcilerModeAnnotation
+		value = config.PodAnnotation
+	}
+
+	r := &PodReconciler {
+		Client: config.Mgr.GetClient(),
+		c:      config,
+		Mode:   mode,
+		Value:  value,
+		Scheme: config.Mgr.GetScheme(),
+	}
+
+	err := ctrl.NewControllerManagedBy(config.Mgr).
+		For(&corev1.Pod{}).
+		Complete(r)
+	if err != nil {
+		return nil, err
+	}
+
+	return r, nil
+}
+
+// Reconcile creates a new SPIFFE ID when pods are created
+func (r *PodReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	var pod corev1.Pod
+	ctx := context.Background()
+
+	if containsString(r.c.DisabledNamespaces, req.NamespacedName.Namespace) {
+		return ctrl.Result{}, nil
+	}
+
+	if err := r.Get(ctx, req.NamespacedName, &pod); err != nil {
+		if !errors.IsNotFound(err) {
+			r.c.Log.WithError(err).Error("unable to fetch Pod")
+			return ctrl.Result{}, err
+		}
+
+		// we'll ignore not-found errors, since they can't be fixed by an immediate
+		// requeue (we'll need to wait for a new notification), and we can get them
+		// on deleted requests.
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	spiffeIdUri := ""
+	switch r.Mode {
+	case PodReconcilerModeServiceAccount:
+		spiffeIdUri = r.makeID("ns/%s/sa/%s", req.Namespace, pod.Spec.ServiceAccountName)
+	case PodReconcilerModeLabel:
+		if val, ok := pod.GetLabels()[r.Value]; ok {
+			spiffeIdUri = r.makeID("%s", val)
+		} else {
+			// No relevant label
+			return ctrl.Result{}, nil
+		}
+	case PodReconcilerModeAnnotation:
+		if val, ok := pod.GetAnnotations()[r.Value]; ok {
+			spiffeIdUri = r.makeID("%s", val)
+		} else {
+			// No relevant annotation
+			return ctrl.Result{}, nil
+		}
+	}
+
+	spiffeId := &spiffeidv1beta1.SpiffeID{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace:   pod.Namespace,
+			Annotations: make(map[string]string),
+		},
+		Spec: spiffeidv1beta1.SpiffeIDSpec{
+			SpiffeId: spiffeIdUri,
+			DnsNames: make([]string, 0),
```

Comment:
This empty slice initialization doesn't seem necessary... (append works on nil slices)

Alternatively, since the pod name seems already available...:
```suggestion
			DnsNames: []string{pod.Name},
```